test-case/check-audio-equalizer.sh

```diff
@@ -44,103 +45,100 @@ func_test_eq()
-    aplay -D$dev -f $fmt -c $channel /dev/zero -d $duration
-    if [ $? -ne 0 ]; then
+    }
+
+    dlogc "$cmd -D $dev -f $fmt -c $channel -r $rate -d $duration $dummy_file"
+    $cmd -D "$dev" -f "$fmt" -c "$channel" -r "$rate" -d "$duration" "$dummy_file" || {
         dloge "Equalizer test failure with $conf"
         return 1
-    fi
+    }
     sleep 1
 }
 
-current_dir=$(dirname ${BASH_SOURCE[0]})
-
-# initialized with negative value. If none of test is performed the test result will be failed.
-result=1
-failed_cnt=0
-
-# TODO: direct import only EQ pipeline and loop for EQ pipeline only
-for idx in $(seq 0 $(expr $PIPELINE_COUNT - 1))
-do
-    dlogi "$idx - total pipeline= $PIPELINE_COUNT"
-    dev=$(func_pipeline_parse_value $idx dev)
-    channel=$(func_pipeline_parse_value $idx channel)
-    rate=$(func_pipeline_parse_value $idx rate)
-    fmt=$(func_pipeline_parse_value $idx fmt)
-    eq_support=$(func_pipeline_parse_value $idx eq)
-    if [ -z "$eq_support" ]; then
-        dlogi "None of FIR/IIF filter is available in this pipeline, skip"
-        continue
-    fi
-    dlogi "eq_support= $eq_support"
-    is_iir=$(echo $eq_support | grep -i iir | wc -l)
-    is_fir=$(echo $eq_support | grep -i fir | wc -l)
-    if [ $is_iir -ne $is_fir ]; then
-        dlogi "no FIR or IIF filter is available in this pipeline, skip"
-        continue
-    fi
-
-    dlogi "0. Get amixer control id for IIR and FIR"
-    IIRid=`amixer -D hw:$sofcard controls | grep EQIIR| head -1| sed 's/numid=\([0-9]*\),.*/\1/'`
-    if [ -z $IIRid ]; then
-            dloge "can't find IIR filter"
-            exit 1
-    fi
-
-    FIRid=`amixer -D hw:$sofcard controls | grep EQFIR| head -1| sed 's/numid=\([0-9]*\),.*/\1/'`
-    if [ -z $FIRid ]; then
-            dloge "can't find FIR filter"
-            exit 1
+# this function performs IIR/FIR filter test
+# param1 must be must be iir or fir
+func_test_filter()
+{
+    local testfilter=$1
+    dlogi "1. Get amixer control id for $testfilter"
+    # TODO: Need to match alsa control id with the filter in the pipeline,
+    #       currently the test discards EQ pipelines except first one.
+    Filterid=$(amixer -D hw:"$sofcard" controls | sed -n -e "/eq${testfilter}/I "'s/numid=\([0-9]*\),.*/\1/p' | head -1)
+    if [ -z "$Filterid" ]; then
+        dloge "can't find $testfilter"
+        exit 1
     fi
 
-    declare -a IIRList=($(ls -d ${current_dir}/eqctl/eq_iir_*.txt))
-    nIIRList=${#IIRList[@]}
-    dlogi "IIR list $nIIRList ${IIRList[*]}"
-    declare -a FIRList=($(ls -d ${current_dir}/eqctl/eq_fir_*.txt))
-    nFIRList=${#FIRList[@]}
-    dlogi "FIR list $nFIRList ${FIRList[*]}"
-    if [ $nIIRList ==  0 ] || [ $nFIRList ==  0 ]; then
-        dloge "IIR or FIR flter coeff list error!"
-	exit 1
+    declare -a FilterList=($(ls -d "${my_dir}"/eqctl/eq_"${testfilter}"_*.txt))
+    nFilterList=${#FilterList[@]}
+    dlogi "2. $testfilter list $nFilterList ${FilterList[*]}"
```

Comment:
I think "$nFilterList ${FilterList[*]}" when check log maybe confuse, need some flag to split it